src/mount/direntry_cache.h

```diff
@@ -474,6 +487,19 @@ class DirEntryCache {
 			++it;
 			erase(entry);
 		}
+
+		// Make sure inode_multiset_ is also clean of outdated entries.
+		// There are scenarios where we can not determine yet the inodes from
+		// the parent, like in unlink
```

Comment:
Consider adding or extending existing Unit tests to checks this corner cases where entries are left hanging in cache and should be removed, or any other possible scenario we are not properly testing as of now...

Reply:
It will be addressed in this task:
https://3.basecamp.com/4375752/buckets/30999151/card_tables/cards/7271353671